app/javascript/src/utils/store.js

```diff
@@ -0,0 +1,8 @@
+export function debounced(cb, ms) {
+  return (values, set) => {
+    const timeout = setTimeout(() => {
+      return set(cb(values))
+    }, ms)
+    return () => clearTimeout(timeout)
+  }
+}
```

Comment:
There actually already is a `debounce` util in `app\javascript\src\debounce.js`. Not sure if that would be suitable to re-use here.

Reply:
I tried it like so
```suggestion
import debounce from "../debounce"

export function debounced(cb, ms) {
  return debounce((values, set) => {
    return set(cb(values))
  }, ms)
}
```

but I was getting type errors because `get(subroutinesMap)` in the linter was `undefined`. I'm guessing there is some race condition somewhere but I don't know where to look.

It's probably better to just use a custom one tailored for `derived`.

Reply:
Oh actually nevermind, that isn't the same of course. The set and return here are very important